### ISSUE TITLE
Deprecated Selectors

### ISSUE THREAD

**Issue by Nikki1993** [atom-material/atom-material-syntax#45]
Latest Atom from master, 1.13.0-dev at the time of writing, reports deprecated selectors for atom editor due to removal of shadow dom.

Starting from Atom v1.13.0, the contents of atom-text-editor elements are no longer encapsulated within a shadow DOM boundary. This means you should stop using :host and ::shadow pseudo-selectors, and prepend all your syntax selectors with syntax--. To prevent breakage with existing style sheets, Atom will automatically upgrade the following selectors:

```
atom-text-editor, :host => atom-text-editor,atom-text-editor
atom-text-editor .gutter, :host .gutter => atom-text-editor .gutter,atom-text-editor .gutter
atom-text-editor .gutter .line-number.cursor-line, :host .gutter .line-number.cursor-line => atom-text-editor .gutter .line-number.cursor-line,atom-text-editor .gutter .line-number.cursor-line
atom-text-editor .gutter .line-number.cursor-line-no-selection, :host .gutter .line-number.cursor-line-no-selection => atom-text-editor .gutter .line-number.cursor-line-no-selection,atom-text-editor .gutter .line-number.cursor-line-no-selection
atom-text-editor .wrap-guide, :host .wrap-guide => atom-text-editor .wrap-guide,atom-text-editor .wrap-guide
atom-text-editor .indent-guide, :host .indent-guide => atom-text-editor .indent-guide,atom-text-editor .indent-guide
atom-text-editor .invisible-character, :host .invisible-character => atom-text-editor .invisible-character,atom-text-editor .invisible-character
atom-text-editor .search-results .marker .region, :host .search-results .marker .region => atom-text-editor .search-results .syntax--marker .region,atom-text-editor .search-results .syntax--marker .region
atom-text-editor .search-results .marker.current-result .region, :host .search-results .marker.current-result .region => atom-text-editor .search-results .syntax--marker.current-result .region,atom-text-editor .search-results .syntax--marker.current-result .region
atom-text-editor.is-focused .cursor, :host(.is-focused) .cursor => atom-text-editor.is-focused .cursor,atom-text-editor .cursor
atom-text-editor.is-focused .selection .region, :host(.is-focused) .selection .region => atom-text-editor.is-focused .selection .region,atom-text-editor .selection .region
atom-text-editor.is-focused .line-number.cursor-line-no-selection, atom-text-editor.is-focused .line.cursor-line, :host(.is-focused) .line-number.cursor-line-no-selection, :host(.is-focused) .line.cursor-line => atom-text-editor.is-focused .line-number.cursor-line-no-selection, atom-text-editor.is-focused .line.cursor-line,atom-text-editor .line-number.cursor-line-no-selection,atom-text-editor .line.cursor-line
.variable.parameter.function => .syntax--variable.syntax--parameter.syntax--function
.comment, .punctuation.definition.comment, .text.cancelled, .meta.punctuation.separator, .markup.strikethrough => .syntax--comment, .syntax--punctuation.syntax--definition.syntax--comment, .syntax--text.cancelled, .syntax--meta.syntax--punctuation.syntax--separator, .syntax--markup.syntax--strikethrough
.notes => .syntax--notes
.text, .source => .syntax--text, .syntax--source
.punctuation.definition.string, .punctuation.definition.string, .punctuation.definition.parameters, .punctuation.definition.string, .punctuation.definition.array => .syntax--punctuation.syntax--definition.syntax--string, .syntax--punctuation.syntax--definition.syntax--string, .syntax--punctuation.syntax--definition.syntax--parameters, .syntax--punctuation.syntax--definition.syntax--string, .syntax--punctuation.syntax--definition.syntax--array
.none => .syntax--none
.keyword.operator => .syntax--keyword.syntax--operator
.keyword => .syntax--keyword
.variable => .syntax--variable
.entity.name.function, .meta.require, .support.function.any-method, .meta.function-call, .support.function, .keyword.other.special-method, .meta.block-level, .meta.function-call.method.with-arguments .variable.function, .source.js .meta.function-call.method.without-arguments.js .variable.function.js => .syntax--entity.syntax--name.syntax--function, .syntax--meta.syntax--require, .syntax--support.syntax--function.syntax--any-method, .syntax--meta.syntax--function-call, .syntax--support.syntax--function, .syntax--keyword.syntax--other.syntax--special-method, .syntax--meta.syntax--block-level, .syntax--meta.syntax--function-call.syntax--method.syntax--with-arguments .syntax--variable.syntax--function, .syntax--source.syntax--js .syntax--meta.syntax--function-call.syntax--method.syntax--without-arguments.syntax--js .syntax--variable.syntax--function.syntax--js
.support.class, .entity.name.class, .entity.name.type.class, .variable.language.this.js => .syntax--support.syntax--class, .syntax--entity.syntax--name.syntax--class, .syntax--entity.syntax--name.syntax--type.syntax--class, .syntax--variable.syntax--language.syntax--this.syntax--js
.meta.class => .syntax--meta.syntax--class
.keyword.other.special-method => .syntax--keyword.syntax--other.syntax--special-method
.storage => .syntax--storage
.support.function, .keyword.operator, .constant.other.color, .meta.tag, .punctuation.definition.tag, .punctuation.separator.inheritance.php, .punctuation.definition.tag.html, .punctuation.definition.tag.begin.html, .punctuation.definition.tag.end.html, .meta.function-call .meta.function-call.arguments .variable.parameter.function, .text.html.markdown .meta.paragraph .meta.link.inline, .text.html.markdown .meta.paragraph .meta.link.inline .punctuation.definition.string.begin.markdown, .text.html.markdown .meta.paragraph .meta.link.inline .punctuation.definition.string.end.markdown, .source.gulpfile.js .variable.other.object.js => .syntax--support.syntax--function, .syntax--keyword.syntax--operator, .syntax--constant.syntax--other.syntax--color, .syntax--meta.syntax--tag, .syntax--punctuation.syntax--definition.syntax--tag, .syntax--punctuation.syntax--separator.syntax--inheritance.syntax--php, .syntax--punctuation.syntax--definition.syntax--tag.syntax--html, .syntax--punctuation.syntax--definition.syntax--tag.syntax--begin.syntax--html, .syntax--punctuation.syntax--definition.syntax--tag.syntax--end.syntax--html, .syntax--meta.syntax--function-call .syntax--meta.syntax--function-call.syntax--arguments .syntax--variable.syntax--parameter.syntax--function, .syntax--text.syntax--html.syntax--markdown .syntax--meta.syntax--paragraph .syntax--meta.syntax--link.syntax--inline, .syntax--text.syntax--html.syntax--markdown .syntax--meta.syntax--paragraph .syntax--meta.syntax--link.syntax--inline .syntax--punctuation.syntax--definition.syntax--string.syntax--begin.syntax--markdown, .syntax--text.syntax--html.syntax--markdown .syntax--meta.syntax--paragraph .syntax--meta.syntax--link.syntax--inline .syntax--punctuation.syntax--definition.syntax--string.syntax--end.syntax--markdown, .syntax--source.gulpfile.syntax--js .syntax--variable.syntax--other.syntax--object.syntax--js
.string, .constant.other.symbol, .entity.other.inherited-class => .syntax--string, .syntax--constant.syntax--other.syntax--symbol, .syntax--entity.syntax--other.syntax--inherited-class
.constant.numeric => .syntax--constant.syntax--numeric
.none => .syntax--none
.meta.function-call .meta.function-call.arguments => .syntax--meta.syntax--function-call .syntax--meta.syntax--function-call.syntax--arguments
.none => .syntax--none
.constant => .syntax--constant
.entity.name.tag, .text.html.basic .meta.tag.other.html .entity.other.attribute-name.html => .syntax--entity.syntax--name.syntax--tag, .syntax--text.syntax--html.syntax--basic .syntax--meta.syntax--tag.syntax--other.syntax--html .syntax--entity.syntax--other.syntax--attribute-name.syntax--html
.entity.other.attribute-name, .meta.tag.any.html .entity.other.attribute-name.html, .text.html.basic .meta.tag.other.html .entity.other.attribute-name.html => .syntax--entity.syntax--other.syntax--attribute-name, .syntax--meta.syntax--tag.syntax--any.syntax--html .syntax--entity.syntax--other.syntax--attribute-name.syntax--html, .syntax--text.syntax--html.syntax--basic .syntax--meta.syntax--tag.syntax--other.syntax--html .syntax--entity.syntax--other.syntax--attribute-name.syntax--html
.entity.other.attribute-name.id => .syntax--entity.syntax--other.syntax--attribute-name.syntax--id
.meta.selector => .syntax--meta.syntax--selector
.none => .syntax--none
.markup.heading .punctuation.definition.heading, .entity.name.section => .syntax--markup.syntax--heading .syntax--punctuation.syntax--definition.syntax--heading, .syntax--entity.syntax--name.syntax--section
.keyword.other.unit => .syntax--keyword.syntax--other.syntax--unit
.markup.bold, .punctuation.definition.bold, .todo.bold => .syntax--markup.syntax--bold, .syntax--punctuation.syntax--definition.syntax--bold, .syntax--todo.syntax--bold
.markup.italic, .punctuation.definition.italic, .todo.italic => .syntax--markup.syntax--italic, .syntax--punctuation.syntax--definition.syntax--italic, .syntax--todo.syntax--italic
.markup.raw.inline => .syntax--markup.syntax--raw.syntax--inline
.string.other.link, .punctuation.definition.string.end.markdown => .syntax--string.syntax--other.syntax--link, .syntax--punctuation.syntax--definition.syntax--string.syntax--end.syntax--markdown
.meta.link => .syntax--meta.syntax--link
.markup.list => .syntax--markup.syntax--list
.markup.quote => .syntax--markup.syntax--quote
.meta.separator => .syntax--meta.syntax--separator
.markup.inserted => .syntax--markup.syntax--inserted
.markup.deleted => .syntax--markup.syntax--deleted
.markup.changed => .syntax--markup.syntax--changed
.constant.other.color, .meta.property-value .support.constant.named-color.css => .syntax--constant.syntax--other.syntax--color, .syntax--meta.syntax--property-value .syntax--support.syntax--constant.named-color.syntax--css
.string.regexp => .syntax--string.syntax--regexp
.constant.character.escape => .syntax--constant.syntax--character.syntax--escape
.punctuation.section.embedded, .variable.interpolation => .syntax--punctuation.syntax--section.syntax--embedded, .syntax--variable.syntax--interpolation
.invalid.illegal => .syntax--invalid.syntax--illegal
.invalid.broken => .syntax--invalid.syntax--broken
.invalid.deprecated => .syntax--invalid.syntax--deprecated
.invalid.unimplemented => .syntax--invalid.syntax--unimplemented
.sublimelinter.outline.illegal => .sublimelinter.outline.syntax--illegal
.sublimelinter.underline.illegal => .sublimelinter.syntax--underline.syntax--illegal
.sublimelinter.outline.warning => .sublimelinter.outline.syntax--warning
.sublimelinter.underline.warning => .sublimelinter.syntax--underline.syntax--warning
.sublimelinter.underline.violation => .sublimelinter.syntax--underline.violation
.sublimelinter.mark.error => .sublimelinter.syntax--mark.syntax--error
.sublimelinter.mark.warning => .sublimelinter.syntax--mark.syntax--warning
.markup.deleted.git_gutter => .syntax--markup.syntax--deleted.git_gutter
.markup.changed.git_gutter => .syntax--markup.syntax--changed.git_gutter
.markup.inserted.git_gutter => .syntax--markup.syntax--inserted.git_gutter
.markup.ignored.git_gutter => .syntax--markup.syntax--ignored.git_gutter
.comment.line.double-slash .punctuation.definition.comment, .meta.structure.array .comment.block.json .punctuation.definition.comment => .syntax--comment.line.syntax--double-slash .syntax--punctuation.syntax--definition.syntax--comment, .syntax--meta.syntax--structure.syntax--array .syntax--comment.syntax--block.syntax--json .syntax--punctuation.syntax--definition.syntax--comment
.text.find-in-files .entity.name.filename.find-in-files => .syntax--text.syntax--find-in-files .syntax--entity.syntax--name.syntax--filename.syntax--find-in-files
.support.type.property-name => .syntax--support.syntax--type.syntax--property-name
.meta.property-list .meta.property-name => .syntax--meta.syntax--property-list .syntax--meta.syntax--property-name
.source.css .keyword.other.unit, .source.less .keyword.other.unit, .source.scss .keyword.other.unit, .source.sass .keyword.other.unit => .syntax--source.syntax--css .syntax--keyword.syntax--other.syntax--unit, .syntax--source.syntax--less .syntax--keyword.syntax--other.syntax--unit, .syntax--source.syntax--scss .syntax--keyword.syntax--other.syntax--unit, .syntax--source.syntax--sass .syntax--keyword.syntax--other.syntax--unit
.entity.other.less.mixin => .syntax--entity.syntax--other.syntax--less.syntax--mixin
.source.less .parameter.less .variable.parameter.misc.css => .syntax--source.syntax--less .syntax--parameter.syntax--less .syntax--variable.syntax--parameter.syntax--misc.syntax--css
.source.less .meta.property-value.css .string.quoted.double.css .comment .markup.raw => .syntax--source.syntax--less .syntax--meta.syntax--property-value.syntax--css .syntax--string.syntax--quoted.syntax--double.syntax--css .syntax--comment .syntax--markup.syntax--raw
.constant.other.color.rgb-value .punctuation.definition.constant => .syntax--constant.syntax--other.syntax--color.syntax--rgb-value .syntax--punctuation.syntax--definition.syntax--constant
.meta.attribute-selector .string.unquoted.attribute-value => .syntax--meta.syntax--attribute-selector .syntax--string.syntax--unquoted.syntax--attribute-value
.meta.attribute-selector .entity.other.attribute-name.attribute => .syntax--meta.syntax--attribute-selector .syntax--entity.syntax--other.syntax--attribute-name.syntax--attribute
.source.gulpfile.js .meta.group.braces.round .meta.group.braces.curly .meta.group.braces.round .meta.function-call.with-arguments.js .variable.function.js, .source.gulpfile.js .meta.group.braces.round .meta.group.braces.curly .variable.function.js => .syntax--source.gulpfile.syntax--js .syntax--meta.syntax--group.syntax--braces.syntax--round .syntax--meta.syntax--group.syntax--braces.syntax--curly .syntax--meta.syntax--group.syntax--braces.syntax--round .syntax--meta.syntax--function-call.syntax--with-arguments.syntax--js .syntax--variable.syntax--function.syntax--js, .syntax--source.gulpfile.syntax--js .syntax--meta.syntax--group.syntax--braces.syntax--round .syntax--meta.syntax--group.syntax--braces.syntax--curly .syntax--variable.syntax--function.syntax--js
.meta.function-call .support.type, .support.type => .syntax--meta.syntax--function-call .syntax--support.syntax--type, .syntax--support.syntax--type
.source.python .meta.function-call.arguments.python => .syntax--source.syntax--python .syntax--meta.syntax--function-call.syntax--arguments.syntax--python
.embedding.php .entity.name.tag => .embedding.syntax--php .syntax--entity.syntax--name.syntax--tag
.source.go .meta.function-call.go => .syntax--source.syntax--go .syntax--meta.syntax--function-call.syntax--go
.meta.group.braces.curly.js .meta.property.object.js .variable.other.object.js => .syntax--meta.syntax--group.syntax--braces.syntax--curly.syntax--js .syntax--meta.syntax--property.syntax--object.syntax--js .syntax--variable.syntax--other.syntax--object.syntax--js
.source.json .meta.structure.dictionary.json .meta.structure.dictionary.value .constant.language => .syntax--source.syntax--json .syntax--meta.syntax--structure.syntax--dictionary.syntax--json .syntax--meta.syntax--structure.syntax--dictionary.syntax--value .syntax--constant.syntax--language
.source.elixir .support.type.elixir, .source.elixir .meta.module.elixir .entity.name.class.elixir => .syntax--source.syntax--elixir .syntax--support.syntax--type.syntax--elixir, .syntax--source.syntax--elixir .syntax--meta.syntax--module.syntax--elixir .syntax--entity.syntax--name.syntax--class.syntax--elixir
.source.elixir .entity.name.function => .syntax--source.syntax--elixir .syntax--entity.syntax--name.syntax--function
.source.elixir .constant.other.symbol.elixir, .source.ruby .constant.other.symbol => .syntax--source.syntax--elixir .syntax--constant.syntax--other.syntax--symbol.syntax--elixir, .syntax--source.syntax--ruby .syntax--constant.syntax--other.syntax--symbol
.source.elixir .punctuation.definition.string => .syntax--source.syntax--elixir .syntax--punctuation.syntax--definition.syntax--string
.source.elixir .variable.other.readwrite.module.elixir, .source.elixir .variable.other.readwrite.module.elixir .punctuation.definition.variable.elixir => .syntax--source.syntax--elixir .syntax--variable.syntax--other.syntax--readwrite.syntax--module.syntax--elixir, .syntax--source.syntax--elixir .syntax--variable.syntax--other.syntax--readwrite.syntax--module.syntax--elixir .syntax--punctuation.syntax--definition.syntax--variable.syntax--elixir
.source.elixir .punctuation.binary.elixir => .syntax--source.syntax--elixir .syntax--punctuation.syntax--binary.syntax--elixir
.support.type.sys-types => .syntax--support.syntax--type.syntax--sys-types
.constant.numeric.line-number.find-in-files:not(.match), .text.find-in-files .constant.numeric.line-number.find-in-files => .syntax--constant.syntax--numeric.line-number.syntax--find-in-files:not(.match), .syntax--text.syntax--find-in-files .syntax--constant.syntax--numeric.line-number.syntax--find-in-files
.entity.name.filename.find-in-files, .text.find-in-files .constant.numeric.line-number.match.find-in-files => .syntax--entity.syntax--name.syntax--filename.syntax--find-in-files, .syntax--text.syntax--find-in-files .syntax--constant.syntax--numeric.line-number.syntax--match.syntax--find-in-files
[data-grammar*="gfm"]::shadow .markup.underline.link => [data-grammar*="gfm"]::shadow .syntax--markup.syntax--underline.syntax--link
[data-grammar*="gfm"]::shadow .markup.underline.link span.punctuation => [data-grammar*="gfm"]::shadow .syntax--markup.syntax--underline.syntax--link span.syntax--punctuation
[data-grammar*="gfm"]::shadow .markup.strike => [data-grammar*="gfm"]::shadow .syntax--markup.syntax--strike
[data-grammar*="gfm"]::shadow .markup.strike::after => [data-grammar*="gfm"]::shadow .syntax--markup.syntax--strike::after
[data-grammar*="gfm"]::shadow .table .border => [data-grammar*="gfm"]::shadow .syntax--table .syntax--border
[data-grammar*="gfm"]::shadow .markup.heading => [data-grammar*="gfm"]::shadow .syntax--markup.syntax--heading
[data-grammar*="gfm"]::shadow .punctuation.definition.begin, [data-grammar*="gfm"]::shadow .punctuation.definition.end => [data-grammar*="gfm"]::shadow .syntax--punctuation.syntax--definition.syntax--begin, [data-grammar*="gfm"]::shadow .syntax--punctuation.syntax--definition.syntax--end
[data-grammar*="gfm"]::shadow .punctuation.definition.begin + span:not(.function.parameter) => [data-grammar*="gfm"]::shadow .syntax--punctuation.syntax--definition.syntax--begin + span:not(.function.parameter)
atom-text-editor.is-focused .selection .region, :host(.is-focused) .selection .region => atom-text-editor.is-focused .selection .region,atom-text-editor .selection .region
.variable.instance, .variable.instance > * => .syntax--variable.syntax--instance, .syntax--variable.syntax--instance > *
.meta.delimiter.period => .syntax--meta.syntax--delimiter.syntax--period
.meta.brace => .syntax--meta.syntax--brace
.meta.tag.doctype => .syntax--meta.syntax--tag.syntax--doctype
.meta .string.quoted => .syntax--meta .syntax--string.syntax--quoted
.meta.array.json .string.quoted => .syntax--meta.syntax--array.syntax--json .syntax--string.syntax--quoted
.meta.array.json .string.quoted > * => .syntax--meta.syntax--array.syntax--json .syntax--string.syntax--quoted > *
.meta .punctuation.separator.key-value + .string.quoted => .syntax--meta .syntax--punctuation.syntax--separator.syntax--key-value + .syntax--string.syntax--quoted
.meta .punctuation.separator.key-value + .string.quoted * => .syntax--meta .syntax--punctuation.syntax--separator.syntax--key-value + .syntax--string.syntax--quoted *
.meta.structure.dictionary .string.quoted => .syntax--meta.syntax--structure.syntax--dictionary .syntax--string.syntax--quoted
.meta.structure.dictionary .string.quoted .punctuation.string => .syntax--meta.syntax--structure.syntax--dictionary .syntax--string.syntax--quoted .syntax--punctuation.syntax--string
.support.constant => .syntax--support.syntax--constant
.keyword.operator.logical, .keyword.operator.comparison, .keyword.operator.assignment => .syntax--keyword.syntax--operator.syntax--logical, .syntax--keyword.syntax--operator.syntax--comparison, .syntax--keyword.syntax--operator.syntax--assignment
.punctuation.section.embedded .source.php, .variable.interpolation .source.php => .syntax--punctuation.syntax--section.syntax--embedded .syntax--source.syntax--php, .syntax--variable.syntax--interpolation .syntax--source.syntax--php
```

Automatic translation of selectors will be removed in a few release cycles to minimize startup time. Please, make sure to upgrade the above selectors as soon as possible.

**Comment by SwankyLynx** [comment]
Yep, I got the same issue after atom being updated to 1.13

**Comment by att1cus** [comment]
Same here.

Atom: 1.13.0
atom-material-syntax: 0.4.6

**Comment by silvestreh** [comment]
Fixed in `0.4.7` 🎉
